src/kdtree.rs

```diff
@@ -35,8 +35,8 @@ impl<'a, T> KdTree<'a, T> {
     }
 
     pub fn new_with_capacity(dimensions: usize, capacity: usize) -> KdTree<'a, T> {
-        let min_bounds = vec![std::f64::NAN; dimensions];
-        let max_bounds = vec![std::f64::NAN; dimensions];
+        let min_bounds = vec![std::f64::INFINITY; dimensions];
+        let max_bounds = vec![std::f64::NEG_INFINITY; dimensions];
```

Comment:
By using ∞ / -∞, we can omit the `.is_nan()` check in the loop. However, this will lead to ∞/-∞ values in zero-dimension trees; a use case that is quite probably non-existant. :smile: